src/com/cft/contactmerge/StreetAddressMatchLogic.java

```diff
@@ -2,13 +2,185 @@
 import java.util.Arrays;
+import java.util.Hashtable;
 
 public abstract class StreetAddressMatchLogic {
+    private static final ArrayList<String> poBoxTypes = new ArrayList<>(Arrays.asList("po", "box"));
+
+    private static final Hashtable<String,String> roadTypes = new Hashtable<String,String>() {{
+        put("st", "street");
+        put("street", "street");
+        put("ave", "avenue");
+        put("avenue", "avenue");
+        put("dr", "drive");
+        put("drive", "drive");
+        put("ln", "lane");
+        put("lane", "lane");
+        put("trl", "trail");
+        put("trail", "trail");
+        put("cir", "circle");
+        put("circle", "circle");
+        put("blvd", "boulevard");
+        put("boulevard", "boulevard");
+        put("rd", "road");
+        put("road", "road");
+    }};
+
+    private static final Hashtable<String,String> compassDirection = new Hashtable<String,String>() {{
+        put("n", "north");
+        put("north", "north");
+        put("s", "south");
+        put("south", "south");
+        put("e", "east");
+        put("east", "east");
+        put("w", "west");
+        put("west", "west");
+    }};
+
+    /*******************************************************************************************************************
+     *************************************************** Get Methods ***************************************************
+     *******************************************************************************************************************/
+    public static Hashtable<String,String> getRoadTypes(){
+        return roadTypes;
+    }
+
+    public static Hashtable<String,String> getCompassDirection(){
+        return compassDirection;
+    }
+
+    public static ArrayList<String> getPoBoxTypes(){
+        return poBoxTypes;
+    }
+
     /*******************************************************************************************************************
      ************************************************* Matching Methods ************************************************
      *******************************************************************************************************************/
-    public static AnswerType getStreetAddressMatchResult(ArrayList<String> stAddress1, ArrayList<String> stAddress2){
-        // ToDo: Fill in code to obtain matching result
+    public static AnswerType getStreetAddressMatchResult(ArrayList<String> stAddressParts1, ArrayList<String> stAddressParts2){
+        // Precondition: All addressParts must follow the rules from the setNormalizeStreetAddress()
+        String st1Direction = "";
+        String st2Direction = "";
+        String st1RoadType = "";
+        String st2RoadType = "";
+
+        // get return on po box addressses (if either one is a po box)
+        if (isAddressPoBox(stAddressParts1) && isAddressPoBox(stAddressParts2)){
+            // Strip all characters out of address1, and only leave numbers
```

Comment:
If your comment is accurate, you can do the same thing with the following code:

        StringBuilder sb = new StringBuilder();
        for (char c: buffer.toCharArray()) { if (Character.isDigit(c)) { sb.append(c); } }

There should be a way to turn this into a single line lambda expression but my first attempt failed at it is getting late.

Reply:
I'll look into this.